Source/Core/Chess.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Generic;
+using Core.Abstractions;
+using Core.Elements;
+
+namespace Core
+{
+    /// <summary>
+    /// Implements <see cref="Chess"/>, and it's basic rules.
+    /// </summary>
+    public class Chess
```

Comment:
This class might be re-implemented as an interface.

The idea is that the chess class will be an abstract class containing all the rules for a chess game. This way, chess variants might be added, such as Fischer 960, or even some simpler rules, such as no castling or no pawn double move.

Reply:
Will be revied later on its onw PR.